runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java

```diff
@@ -449,14 +499,27 @@ private void killedOne()
 		amount--;
 		config.amount(amount); // save changed value
 
-		if (!config.showInfobox())
+		if (config.showInfobox())
 		{
-			return;
+			// add and update counter, set timer
+			addCounter();
+			counter.setText(String.valueOf(amount));
+		}
+
+		if (config.showBraceletInfoBox())
+		{
+			if (hasExpeditiousEquip || hasExpeditiousInvent)
+			{
+				addExpeditiousCounter();
+				expeditiousCounter.setText(String.valueOf(expeditiousChargeCount));
```

Comment:
The charges are not changed here... instead it should update the infobox when it does change